packages/google_maps_flutter/google_maps_flutter_ios/ios/Classes/GoogleMapMarkerController.m

```diff
@@ -196,20 +201,35 @@ - (void)changeMarkers:(NSArray<FGMPlatformMarker *> *)markersToChange {
 
 - (void)changeMarker:(FGMPlatformMarker *)markerToChange {
   NSString *markerIdentifier = markerToChange.markerId;
-  NSString *clusterManagerIdentifier = markerToChange.clusterManagerId;
 
   FLTGoogleMapMarkerController *controller = self.markerIdentifierToController[markerIdentifier];
   if (!controller) {
     return;
   }
+
+  NSString *clusterManagerIdentifier = markerToChange.clusterManagerId;
   NSString *previousClusterManagerIdentifier = [controller clusterManagerIdentifier];
-  if (![previousClusterManagerIdentifier isEqualToString:clusterManagerIdentifier]) {
-    [self removeMarker:markerIdentifier];
-    [self addMarker:markerToChange];
-  } else {
-    [controller updateFromPlatformMarker:markerToChange
-                               registrar:self.registrar
-                             screenScale:[self getScreenScale]];
+  [controller updateFromPlatformMarker:markerToChange
+                             registrar:self.registrar
+                           screenScale:[self getScreenScale]];
+
+  if ([controller.marker conformsToProtocol:@protocol(GMUClusterItem)]) {
+    if (previousClusterManagerIdentifier &&
+        ![previousClusterManagerIdentifier isEqualToString:clusterManagerIdentifier]) {
```

Comment:
`if (previousClusterManagerIdentifier && ![clusterManagerIdentifier isEqualToString:previousClusterManagerIdentifier])`

In general, a lot of these kind of framework methods don't handle nil arguments, so we shouldn't pass in a potentially `nil` `clusterManagerIdentifier`). This ordering will end up with `[nil isEqualToString:<a value we've established as non-nil>]`, which gives us the intended result of `NO` without passing `nil` as an argument.

Reply:
Fixed